File: library/src/main/java/no/wtw/android/architectureutils/externalpayment/ExternalPaymentWebViewListener.java
package no.wtw.android.architectureutils.externalpayment;

public interface ExternalPaymentWebViewListener {

    void onExternalPaymentCancelled();

    void onExternalPaymentSuccess(String paymentMethodId);

    void setIsLoading(boolean isLoading);

    String getExternalPaymentCancelledUrl();

    String getExternalPaymentSuccessUrl();

    /* Return true for exact match on cancel URL, or false for startsWith-match */
    boolean isExternalPaymentCancelledUrlExactMatch();

    /* Return true for exact match on success URL, or false for startsWith-match */
    boolean isExternalPaymentSuccessUrlExactMatch();

}

File: library/src/main/java/no/wtw/android/architectureutils/externalpayment/ExternalPaymentWebViewListener.kt
package no.wtw.android.architectureutils.externalpayment

interface ExternalPaymentWebViewListener {

    fun onExternalPaymentCancelled()
    fun onExternalPaymentSuccess(paymentMethodId: String)
    fun setIsLoading(isLoading: Boolean)
    fun getExternalPaymentCancelledUrl(): String
    fun getExternalPaymentSuccessUrl(): String

    /* Return true for exact match on cancel URL, or false for startsWith-match */
    fun isExternalPaymentCancelledUrlExactMatch(): Boolean
    /* Return true for exact match on success URL, or false for startsWith-match */
    fun isExternalPaymentSuccessUrlExactMatch(): Boolean

    fun handleSpecialUrl(url: String): Boolean

}